### PR TITLE
checksrc: check for spaces around '?', '>' and '<'

### PR DESCRIPTION
**Pull request by bagder**
Inspired by #14912

### CODE REVIEW

**Comment by dfandrich** [comment]
Analysis of PR #14921 at [07ffe6ff](https://github.com/curl/curl/commit/07ffe6ffb51fb37575141e7cabbece2f09225c53):

[Test 1185 failed](https://circleci.com/gh/curl/curl/107004), which has NOT been flaky recently, so **there could be a real issue in the PR**.  Note that this test has failed in 94 different CI jobs (the link just goes to one of them).

###### Generated by [Testclutch](https://testclutch.curl.se/)

File: tests/server/util.c [hunk 1]
```diff
@@ -671,7 +671,7 @@ static SIGHANDLER_T set_signal(int signum, SIGHANDLER_T handler,
   sa.sa_handler = handler;
   sigemptyset(&sa.sa_mask);
   sigaddset(&sa.sa_mask, signum);
-  sa.sa_flags = restartable? SA_RESTART: 0;
+  sa.sa_flags = restartable ? SA_RESTART: 0;
```
Comment (MarcelRaad) [inline tests/server/util.c:674]: These asymmetric ones look strange to me now, where there are spaces on both sides of `?` but only on one side of `:`. Looks good to me other than that.

Comment (bagder) [reply tests/server/util.c:674]: It's not intentional, but checksrc does not check the colons (yet).

**Comment by bagder** [comment]
Verifying spaces around the colons turns out to be quite difficult without catching a few false positives: they are used for switches, goto labels and bit-size indicators in struct declarations where we do not want to insist on use of space. So I'll not add that check to checksrc in this PR but I used an attempt to find a bunch of occurrences fixed here.

**Comment by bagder** [comment]
Buckle up!

**Pull request closed without merging**